dungeon/src/contrib/utils/DynamicCompiler.java

```diff
@@ -0,0 +1,153 @@
+    File outputFile = new File(outputRoot, className.replace('.', '/') + ".java");
+    outputFile.getParentFile().mkdirs();
+
+    Path filePath = Paths.get(sourcePath.pathString());
+    String sourceCode = Files.readString(filePath);
+
+    try (BufferedWriter writer = new BufferedWriter(new FileWriter(outputFile))) {
+      writer.write(sourceCode);
+    }
+
+    JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
+    StandardJavaFileManager fileManager = compiler.getStandardFileManager(null, null, null);
+
+    Iterable<? extends JavaFileObject> compilationUnits =
+        fileManager.getJavaFileObjects(outputFile);
+    JavaCompiler.CompilationTask task =
+        compiler.getTask(null, fileManager, null, null, null, compilationUnits);
+
+    boolean success = task.call();
+    if (!success) {
+      throw new Exception("Compilation failed.");
+    }
+
+    // Load compiled class using custom loader to override any previous versions
+    MyClassLoader classLoader = new MyClassLoader(new URL[] {outputRoot.toURI().toURL()});
+    Class<?> loadedClass = classLoader.loadClass(className);
+    outputFile.delete();
+
+    return loadedClass;
+  }
+
+  /**
+   * Dynamically instantiates a class from a source file using an optional set of constructor
+   * arguments.
+   *
+   * <p>This method compiles the provided Java source file, loads the resulting class using a custom
+   * class loader, and creates a new instance using the specified constructor arguments.
+   *
+   * @param sourcePath Path to the Java source file.
+   * @param className Fully qualified name of the class to load (including package name). This must
+   *     exactly match the package and class declaration inside the file.
+   * @param args Optional constructor arguments as {@code Tuple<Class<?>, Object>}: each tuple
+   *     specifies the expected parameter type and the value to pass. The order of the arguments
+   *     must match the parameter order in the target constructor.
+   * @return A new instance of the loaded and instantiated class.
+   * @throws Exception If compilation, class loading, or instantiation fails.
+   *     <strong>Important:</strong> The target class must contain a constructor that exactly
+   *     matches the parameter types and order given in {@code args}. Otherwise, a {@code
+   *     NoSuchMethodException} will be thrown.
+   */
+  @SafeVarargs
+  public static Object loadUserInstance(
+      IPath sourcePath, String className, Tuple<Class<?>, Object>... args) throws Exception {
+    if (args == null) throw new IllegalArgumentException("Args can not be null");
+
+    Class<?> newClass = DynamicCompiler.compileAndLoad(sourcePath, className);
+    Class<?>[] paramTypes = Arrays.stream(args).map(Tuple::a).toArray(Class[]::new);
+    Object[] paramValues = Arrays.stream(args).map(Tuple::b).toArray();
+
+    Constructor<?> ctor = newClass.getConstructor(paramTypes);
+    return ctor.newInstance(paramValues);
+  }
```

Comment:
```suggestion
  public static <T> T loadUserInstance(
    IPath sourcePath,
    String className,
    Class<T> expectedType,
    Tuple<Class<?>, Object>... args
  ) throws Exception {
    if (args == null) args = new Tuple[0];
  
    Class<?> compiledClass = DynamicCompiler.compileAndLoad(sourcePath, className);
    Class<?>[] paramTypes  = Arrays.stream(args).map(Tuple::a).toArray(Class[]::new);
    Object[]    paramValues = Arrays.stream(args).map(Tuple::b).toArray();
  
    Constructor<?> ctor = compiledClass.getConstructor(paramTypes);
    Object instance     = ctor.newInstance(paramValues);
  
    if (!expectedType.isInstance(instance)) {
      throw new ClassCastException("Loaded class is not of type " + expectedType.getName());
    }
  
    return expectedType.cast(instance);
  }
```
So müsste man nicht mehr selber casten.
Aufruf wäre dann:
```java
MyType instance = loadUserInstance(
    sourcePath,
    "com.example.MyClass",
    MyType.class,
    new Tuple<>(String.class, "hello"),
    new Tuple<>(int.class, 42)
);
```
Und hier könnte man dann evtl. dann auch sourcePath und className verbinden wie wir das grade oben diskutieren

Reply:
hmmm. das geht sogar noch schöner:

```java
public static <T> T loadUserInstance2(Class<? extends T> dynClasz, Tuple<Class<?>, Object>... args) throws Exception {
        if (args == null) args = new Tuple[0];

        Class<?> compiledClass = DynamicCompiler.compileAndLoad(() -> "src/" + dynClasz.getName().replace(".", "/") + ".java", dynClasz.getName());
        Class<?>[] paramTypes = Arrays.stream(args).map(Tuple::a).toArray(Class[]::new);
        Object[] paramValues = Arrays.stream(args).map(Tuple::b).toArray();

        Constructor<?> ctor = compiledClass.getConstructor(paramTypes);
        Object instance = ctor.newInstance(paramValues);

        return ((T) instance);
    }
```

und der aufruf dann

```java
Wuppie x = DynamicCompiler.loadUserInstance2(MyClass.class);
Wuppie x = DynamicCompiler.<Wuppie>loadUserInstance2(MyClass.class);
```

Reply:
#1962